### PR TITLE
MinSafeTS support TiFlash

### PR DESCRIPTION
**Pull request by hehechen**
Signed-off-by: hehechen <awd123456sss@gmail.com>
MinSafeTS support TiFlash to support TiFlash stale read.

### CODE REVIEW

File: tikv/kv.go [hunk 1]
```diff
@@ -522,12 +522,17 @@ func (s *KVStore) safeTSUpdater() {
 
 func (s *KVStore) updateSafeTS(ctx context.Context) {
 	stores := s.regionCache.GetStoresByType(tikvrpc.TiKV)
+	tiflashStores := s.regionCache.GetStoresByType(tikvrpc.TiFlash)
+	stores = append(stores, tiflashStores...)
```
Comment (JmPotato) [inline tikv/kv.go:526]: What about adding a method named `GetStores` all `GetAllStores`?

**Review by JmPotato — COMMENTED** [review]
Could we add a test for this change also? The rest LGTM.

File: internal/locate/region_cache.go [hunk 2]
```diff
@@ -2285,7 +2289,7 @@ func (s *Store) reResolve(c *RegionCache) (bool, error) {
 	storeType := tikvrpc.GetStoreTypeByMeta(store)
 	addr = store.GetAddress()
 	if s.addr != addr || !s.IsSameLabels(store.GetLabels()) {
-		newStore := &Store{storeID: s.storeID, addr: addr, saddr: store.GetStatusAddress(), storeType: storeType, labels: store.GetLabels(), state: uint64(resolved)}
+		newStore := &Store{storeID: s.storeID, addr: addr, peerAddr: store.GetPeerAddress(), saddr: store.GetStatusAddress(), storeType: storeType, labels: store.GetLabels(), state: uint64(resolved)}
```
Comment (lhy1024) [inline internal/locate/region_cache.go:2292]: Add peer addr will make MinSafeTS support TiFlash？

**Review by lhy1024 — COMMENTED** [review]
Maybe we need to add some tests

File: internal/locate/region_cache.go (hunk 2 above)
Comment (hehechen) [reply internal/locate/region_cache.go:2292]: > Add peer addr will make MinSafeTS support TiFlash？

client-go need to get `StoreSafeTS` from TiFlash by sending GRPC message to `peer addr` of TiFlash. Can see the modification in `updateSafeTS`.

**Review by lhy1024 — APPROVED** [review]

**Review by sticnarf — APPROVED** [review]
LGTM

**Comment by sticnarf** [comment]
@hehechen There are race and lint errors.

**Comment by sticnarf** [comment]
There is still a lint error remaining.

You can ignore the integration test failure. It's just flaky.

**Comment by hehechen** [comment]
> There is still a lint error remaining.
> 
> You can ignore the integration test failure. It's just flaky.

Done. Checked in local machine by `golangci-lint run`.

**Pull request merged**